README.md

```diff
@@ -1,29 +1,82 @@
 
-* Authentication between services is already established. This means that this specification doesn't cover the way a service authenticates incoming API calls (e.g. through an API Key, VPN connection or IP whitelisting). In this scope we assume that the services are already authenticated.
 * Provider knows the consumer (both endpoint and user) when it creates a share with the consumer (also see [#26](https://github.com/cs3org/OCM-API/issues/26)). How this is known is not part of this spec.
 * Consumer doesn't have to accept a share, the resource will be available to the consumer immediately (#25).
 * Dealing with incoming shares is a vendor specific implementation. One vendor might use an 'accept before' process while another vendor might use a 'decline after' approach. This is considered part of the UX and thus not part of the interaction between different vendors. However, the consumer could notify the provider by using the introduced `/notifications` endpoint (also see [#27](https://github.com/cs3org/OCM-API/issues/27)).
 * Reverting access to outgoing shares is a vendor specific implementation. One vendor might delete an entire share while another might invalidate an access token. This is considered part of vendor specific internals and thus not part of the interaction between different vendors. However, the provider could notify the consumer by using the introduced `/notifications` endpoint (also see [#27](https://github.com/cs3org/OCM-API/issues/27)).
 * The actual file sync isn't a part of this specification. To keep this specification 'future proof', the file sync protocol will be embedded as a separate object in Open Cloud Mesh API calls. This protocol object contains all protocol specific options, e.g. WebDAV specific options.
 
-## API Documentation
+
+## Specification
+### Discovery
+Authentication between services is already established. This means that this specification doesn't cover the way a service authenticates incoming API calls (e.g. through an API Key, VPN connection or IP whitelisting). In this scope we assume that the services are already authenticated.
+
+If a finite whitelist of receiver servers exists on the sender
+side, then this list may already contain all necessary endpoint details.
+
+When a sending server allows sending to any internet-hosted receiving server, then discovery can happen from the sharee address, using the `/ocm-provider` well-known URL that receiving servers MAY provide. For instance, if `alice at sender.com` wants to share a resource with `bob at receiver.com` then a GET call can be made to
+`https://receiver.com/ocm-provider` to which the receiving server MAY respond with
+a JSON document containing for instance:
+```json
+{
+  "enabled":true,
+  "apiVersion":"1.0-proposal1",
+  "endPoint": "https://receiver.com/api/ocm",
+  "resourceTypes": [
+    {
+      "name":"file",
+      "shareTypes": ["user", "group"],
+      "protocols": {
+        "webdav": (any string)
+      }
+    }
+  ]
+}
+```
```

Comment:
```suggestion
When a sending server allows sending to any internet-hosted receiving server, then discovery can happen from the sharee address, using the `/ocm-provider` well-known URL that receiving servers MAY provide according to this [specification](https://cs3org.github.io/OCM-API/docs.html?branch=develop&repo=OCM-API&user=cs3org#/paths/~1ocm-provider/get).
```

Reply:
(here I suggested to remove the example payload because the `/ocm-provider` endpoint is now into the spec)